backend/sidekiq_config.md

```diff
@@ -0,0 +1,130 @@
+##Background Jobs using Sidekiq
+
+This document outlines the configuration required for a Rails application using the Sidekiq gem for background job processing to be deployed on Heroku.
+
+For detailed information, refer to:
+
+
+* [https://github.com/mperham/sidekiq/wiki/Getting-Started]()
+* [https://devcenter.heroku.com/articles/redistogo]()
+* [https://devcenter.heroku.com/articles/scheduler]()
+
+
+###Rails Configuration
+
+Include the sidekiq gem in the ```Gemfile``` and run install bundle.
+
+```
+  gem ‘sidekiq’
+```
+
+Add the following two lines in the ```app/Procfile```.
+
+```
+  web: bundle exec puma -p $PORT -C ./config/puma.rb
+  worker: bundle exec sidekiq -C config/sidekiq.yml
+```
+
+Add the configuration line for sidekiq in the
+```config/application.rb```.
+
+```
+module AppName
+  class Application < Rails::Application
+    ...
+    config.active_job.queue_adapter = :sidekiq
+  end
+end
+```
+
+Add a sidekiq initializer: ```config/initializers/sidekiq.rb```
+
+```
+# config/initializers/sidekiq.rb
+#
+# Perform Sidekiq jobs immediately in development,
+# so you don't have to run a separate process.
+# You'll also benefit from code reloading.
+if Rails.env.development?
+  require 'sidekiq/testing'
+  Sidekiq::Testing.inline!
+end
+```
+
+Add a sidekiq yml file: ```config/sidekiq.yml```
+
+```
+---
+:verbose: true
+:namespace: sidekiq
+:concurrency: <%= ENV.fetch('SIDEKIQ_CONCURRENCY', 25) %> (3 for Heroku)
```

Comment:
I'd remove `(3 for Heroku)` here and add it above the code block lest someone blindly copy paste this. Which we would of course never do :D